bip-???-cat.mediawiki

```diff
@@ -0,0 +1,91 @@
+<pre>
+  BIP: ???
+  Layer: Consensus (soft fork)
+  Title: OP_CAT
+  Author: Ethan Heilman <ethan.r.heilman@gmail.com>
+          Armin Sabouri <arminsdev@gmail.com>
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-10-21
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-op-cat
+  License: BSD-3-Clause
+</pre>
+
+==Abstract==
+
+This BIP reintroduces OP_CAT in the form of a new tapscript opcode which allows the concatenation of two values on the stack. This opcode would be activated via a soft fork by redefining the opcode OP_SUCCESS126.
+
+When evaluated the OP_CAT instruction:
+# Pops the top two values off the stack,
+# concatenates the popped values together,
+# and then pushes the concatenated value on the top of the stack.
+
+OP_CAT fails if there are less than two values on the stack or if a concatenated value would have a combined size greater than the maximum script element size of 520 Bytes.
+
+==Motivation==
+Bitcoin tapscript lacks a general purpose way of combining objects on the stack restricting the expressiveness and power of tapscript. This prevents among many other things the ability to construct and evaluate merkle trees and other hashed data structures in tapscript. OP_CAT by adding a general purpose way to concatenate stack values would overcome this limitation and greatly increase the functionality of tapscript.
```

Comment:
"Bitcoin Script" or just "tapscript" are perfectly normal terms, their combination somewhat less so:
```suggestion
Tapscript lacks a general purpose way of combining objects on the stack restricting the expressiveness and power of tapscript. This prevents among many other things the ability to construct and evaluate merkle trees and other hashed data structures in tapscript. OP_CAT by adding a general purpose way to concatenate stack values would overcome this limitation and greatly increase the functionality of tapscript.
```

Reply:
I think the original is fine.

Reply:
Yes, I've checked and it seems "Bitcoin tapscript" isn't actually all that uncommon. Disregard.